roles/ansible-test-splitter/files/split_targets.py

```diff
@@ -30,16 +30,35 @@ def is_slow(lines):
     return False
 
 
-def iter_target_dir(collection_path):
-    targets = PosixPath(os.path.join(collection_path, "tests/integration/targets/"))
-    for target in targets.glob("*"):
+def get_all_targets(collection_path):
+    """Return all the targets from a directory."""
+    targets_dir = PosixPath(os.path.join(collection_path, "tests/integration/targets/"))
+    targets = {}
+    for target in targets_dir.glob("*"):
         aliases = target / "aliases"
         if not target.is_dir():
             continue
         if not aliases.is_file():
             continue
         lines = aliases.read_text().split("\n")
-        yield (target, lines)
+        targets[target.name] = lines
+    return targets
+
+
+def evaluate_redirection(in_targets):
```

Comment:
Actually, we should take `args.targets` as a parameter and use the value to decide what target we keep.